substrate-node/pallets/pallet-smart-contract/src/lib.rs

```diff
@@ -115,110 +185,127 @@ decl_error! {
         NodeIsNotDedicated,
         NodeNotAvailableToDeploy,
         CannotUpdateContractInGraceState,
-        NumOverflow
+        NumOverflow,
     }
-}
-
-decl_storage! {
-    trait Store for Module<T: Config> as SmartContractModule {
-        pub Contracts get(fn contracts): map hasher(blake2_128_concat) u64 => types::Contract;
-        pub ContractBillingInformationByID get(fn contract_billing_information_by_id): map hasher(blake2_128_concat) u64 => types::ContractBillingInformation;
-        pub NodeContractResources get(fn node_contract_resources): map hasher(blake2_128_concat) u64 => types::ContractResources;
-
-        // ContractIDByNodeIDAndHash is a mapping for a contract ID by supplying a node_id and a deployment_hash
-        // this combination makes a deployment for a user / node unique
-        pub ContractIDByNodeIDAndHash get(fn node_contract_by_hash): double_map hasher(blake2_128_concat) u32, hasher(blake2_128_concat) Vec<u8> => u64;
-
-        pub ActiveNodeContracts get(fn active_node_contracts): map hasher(blake2_128_concat) u32 => Vec<u64>;
-        pub ContractsToBillAt get(fn contract_to_bill_at_block): map hasher(blake2_128_concat) u64 => Vec<u64>;
-        pub ContractLock get(fn contract_number_of_cylces_billed): map hasher(blake2_128_concat) u64 => types::ContractLock<BalanceOf<T>>;
-        pub ContractIDByNameRegistration get(fn contract_id_by_name_registration): map hasher(blake2_128_concat) Vec<u8> => u64;
-        pub ActiveRentContractForNode get(fn active_rent_contracts): map hasher(blake2_128_concat) u32 => types::Contract;
 
-        // ID maps
-        pub ContractID: u64;
-    }
-}
-
-decl_module! {
-    pub struct Module<T: Config> for enum Call where origin: T::Origin {
-        fn deposit_event() = default;
-
-        #[weight = 100_000_000]
-        fn create_node_contract(origin, node_id: u32, data: Vec<u8>, deployment_hash: Vec<u8>, public_ips: u32){
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        #[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
+        pub fn create_node_contract(
+            origin: OriginFor<T>,
+            node_id: u32,
+            data: Vec<u8>,
+            deployment_hash: Vec<u8>,
```

Comment:
We should check which hash functions are used by other tools. Essentially hash output has a fixed size, so rather than an arbitrary length Vec, this can be replaced by a (stack allocated) array, which means you have a lenght check as well as a slight performance improvement as no allocation needs to happen at runtime

Reply:
@xmonader what is the size of the deployment hash? Which hash function is used by our tools?